benchmark/tools/inference_workload_matgen.py

```diff
@@ -0,0 +1,423 @@
+            curr_mem = 0
+    if curr:
+        batches.append(Batch(user_requests=curr))
+        print(f"Last batch is incomplete, his size is {len(curr)}")
+
+    return batches
+
+def gen_matrices_and_compute_time(
+    batches: List[Batch],
+    prefill_workers: List[List[int]],
+    decode_workers: List[List[int]],
+    model_config: ModelConfig,
+    prefill_worker_config: WorkerConfig,
+    decode_worker_config: WorkerConfig,
+) -> List[TransferMatrix]:
+    """
+    Args:
+        - batches: List of batches
+    """
+    # For now, every prefill worker is bound to a single decode worker
+    assert len(prefill_workers) == len(decode_workers), f"Prefill and decode workers must have the same number of workers, got {len(prefill_workers)} and {len(decode_workers)}"
+
+    # Assertions
+    all_ranks = list(r for worker in prefill_workers + decode_workers for r in worker)
+    world_size = max(all_ranks) + 1
+    assert set(all_ranks) == set(range(world_size)), "Ranks are missing"
+
+    workers_coupling = list(zip(prefill_workers, decode_workers))
+
+    workers_pool = cycle(workers_coupling)
+    matrices = []
+
+    for batch in tqdm(batches, desc="Generating matrices"): 
+
+        prefill_worker, decode_worker = next(workers_pool)
+        mat = gen_matrix(batch, world_size, prefill_worker, decode_worker, model_config, prefill_worker_config, decode_worker_config)
+
+        compute_time = estimate_compute_time(batch, model_config)
+        matrix_obj = TransferMatrix(matrix=mat, compute_time=compute_time, isl=batch.total_isl)
+        matrices.append(matrix_obj)
+
+
+    return matrices
+
+def gen_matrix(
+    batch: Batch,
+    world_size: int,
+    prefill_worker: List[int],
+    decode_worker: List[int],
+    model_config: ModelConfig,
+    prefill_worker_config: WorkerConfig,
+    decode_worker_config: WorkerConfig,
+):
+    kv_size = batch.kv_cache_size(model_config) 
+    kv_slice_size = kv_size / prefill_worker_config.tp / prefill_worker_config.pp / prefill_worker_config.cp
+
+    num_peers = decode_worker_config.tp / prefill_worker_config.tp / prefill_worker_config.cp
+    if num_peers % 1 != 0:
+        raise ValueError("Prefill TP*Prefill CP must be a divisor of decode TP")
+    num_peers = int(num_peers)
+    buf_size = kv_slice_size / num_peers
+
+    #print(f"kv_size: {format_size(kv_size)}, kv_slice_size: {format_size(kv_slice_size)}, buf_size: {format_size(buf_size)}, num_peers: {num_peers}")
+
+    mat = np.zeros((world_size, world_size))
+
+    dst_iter = iter(decode_worker)
+    for rank in prefill_worker:
+        for _ in range(num_peers):
+            dst = next(dst_iter)
+            mat[rank, dst] = buf_size
+    return mat
+
+def estimate_compute_time(
+    batch: Batch,
+    model_config: ModelConfig,
+    flops = 36*1E12, # 36 TFlops (h100)
+):
+    """Estimate the compute time of a batch, in seconds"""
+    flop = 2 * batch.size * model_config.num_layers * model_config.hidden_size * batch.total_isl**2
```

Comment:
* Why is total_isl squared?
* Does the multiplier 2 come from KV?
It's better to explain the formulation in comments.